components/camera/transformpipeline/segmenter.go

```diff
@@ -0,0 +1,123 @@
+package transformpipeline
+
+import (
+	"context"
+	"fmt"
+	"image"
+
+	"github.com/viamrobotics/gostream"
+	"go.opencensus.io/trace"
+	goutils "go.viam.com/utils"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+	"go.viam.com/rdk/services/vision"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+// segmenterConfig is the attribute struct for segementers (their name as found in the vision service).
+type segmenterConfig struct {
+	SegmenterName string `json:"segmenter_name"`
+	CameraName    string `json:"camera_name"`
```

Comment:
Don't put the camera name in the Config, but take it from the original pipeline source. So feed it in as a parameter in `newSegmentationsTransform`